javascript.js

```diff
@@ -0,0 +1,188 @@
+//store data in local storage through this function
+function fetchLocalStorageData() {
+  const experienceList = document.getElementById("experienceList");
+  experienceList.innerHTML = ""; // Clear previous content
+
+  const localData = JSON.parse(localStorage.getItem("experiencedata")) || [];
+  localData.forEach((singleExperience) => {
+    const experienceEntry = document.createElement("div");
+    experienceEntry.classList.add("experience-entry");
+
+    //edit entity start over  here
+    const editButton = document.createElement("button");
+    editButton.innerText = "edit";
+    editButton.addEventListener("click", () =>
+      editEntry(singleExperience, experienceEntry)
+    );
+
+    //edit function end here
+
+    //delete entity here
+    const deleteButton = document.createElement("button"); // create a delete button and assign variable to the
+    deleteButton.innerText = "Delete"; // inner text is delete button inside the button
+    deleteButton.addEventListener("click", () =>
+      deleteEntry(singleExperience.id)
+    );
+
+    experienceEntry.innerHTML = `
+      <h2><strong>${singleExperience.companyName}</strong><br></h2>
+      <p>${singleExperience.startDate} - ${singleExperience.endDate} </p>
+      <p>${singleExperience.description}</p>     
+    `;
+
+    experienceEntry.appendChild(editButton);
+    experienceEntry.appendChild(deleteButton);
+    experienceList.appendChild(experienceEntry);
+  });
+}
```

Comment:
you have misunderstood my last comment. i want you to change the function name such that it reflects its actual purpose...

you need to refactor this whole function. a function should have a single reponsibility, but your function is doing mulltiple things. 

- it is fetching experience items from local-storage `(function fetchExperienceList())`
- adding eventListeners for edit button click event
- creating an edit button and adding it to DOM (which should be in the markup and just the event listener should be added here in .js file)
- rendering the experinceList on DOM `(function renderExperienceList())`

thats why you are having trouble naming it. refactor the whole function into simpler multiple functions with appropriate names as suggested

Reply:
resolved!